geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANPropagationLoopBackDUnitTest.java

```diff
@@ -433,6 +455,264 @@ public void unstartedSenderShouldNotAddReceivedEventsIntoTmpDropped() throws Exc
+    vm2.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_PR", 100));
+    vm4.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_PR", 100));
+  }
+
+  /**
+   * Test loop back issue between 2 WAN sites (LN & NY). LN -> NY -> LN.
+   * Site-LN: dsid=2: senderId="ny": vm2, vm4
+   * Site-NY: dsid=1: senderId="ln": vm3, vm5
+   * NY site's sender's manual-start=true
+   *
+   * Stop the sender in LN
+   * put some events from LN immediately, they will be added to tmpDroppedEvents
+   * Restart the sender in LN, and make sure the events in tmpDroppedEvents are sent to NY.
+   */
+  @Test
+  public void restartedSenderShouldAddReceivedEventsIntoTmpDropped()
+      throws Exception {
+    Integer lnPort = (Integer) vm0.invoke(() -> WANTestBase.createFirstLocatorWithDSId(2));
+    Integer nyPort = (Integer) vm1.invoke(() -> WANTestBase.createFirstRemoteLocator(1, lnPort));
+
+    // create receiver on site-ln and site-ny
+    createCacheInVMs(lnPort, vm2, vm4);
+    createReceiverInVMs(vm2, vm4);
+    createCacheInVMs(nyPort, vm3, vm5);
+    createReceiverInVMs(vm3, vm5);
+
+    // create senders on site-ln, Note: sender-id is its destination, i.e. ny
+    vm2.invoke(() -> WANTestBase.createSender("ny", 1, true, 100, 10, false, false, null, true));
+    vm4.invoke(() -> WANTestBase.createSender("ny", 1, true, 100, 10, false, false, null, true));
+
+    // create senders on site-ny, Note: sender-id is its destination, i.e. ln
+    vm3.invoke(() -> WANTestBase.createSender("ln", 2, true, 100, 10, false, false, null, true));
+    vm5.invoke(() -> WANTestBase.createSender("ln", 2, true, 100, 10, false, false, null, true));
+
+    // create PR on site-ln
+    vm2.invoke(() -> WANTestBase.createPartitionedRegion(getTestMethodName() + "_PR", "ny", 1, 100,
+        isOffHeap()));
+    vm4.invoke(() -> WANTestBase.createPartitionedRegion(getTestMethodName() + "_PR", "ny", 1, 100,
+        isOffHeap()));
+
+    // create PR on site-ny
+    vm3.invoke(() -> WANTestBase.createPartitionedRegion(getTestMethodName() + "_PR", "ln", 1, 100,
+        isOffHeap()));
+    vm5.invoke(() -> WANTestBase.createPartitionedRegion(getTestMethodName() + "_PR", "ln", 1, 100,
+        isOffHeap()));
+
+    // start sender on site-ln
+    startSenderInVMs("ny", vm2, vm4);
+
+    // do 100 puts on site-ln
+    vm2.invoke(() -> WANTestBase.doPutsFrom(getTestMethodName() + "_PR", 0, 100));
+
+    // verify site-ny has received the events from site-ln
+    vm3.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_PR", 100));
+    vm5.invoke(() -> WANTestBase.validateRegionSize(getTestMethodName() + "_PR", 100));
+
+    // verify tmpDroppedEvents is 0 now at site-ln
+    vm2.invoke(() -> WANTestBase.verifyTmpDroppedEventSize("ny", 0));
+    vm4.invoke(() -> WANTestBase.verifyTmpDroppedEventSize("ny", 0));
+
+    // stop sender on site-ny
+    vm2.invoke(() -> stopSender("ny"));
+    vm4.invoke(() -> stopSender("ny"));
```

Comment:
I thought the test should somehow just stop just the primary gatewaysender, not the secondary. and sleep for 15 minutes, do some puts, and make sure events in secondary sender is all drained....

Reply:
If we stop just the primary, wait for 15 seconds and then do the puts, the primary will not longer store the dropped events and when it is started again will not send the events to the secondary to drain them from the queues.
That test case will not work with this solution.
The solution tries to solve the problem when you stop all gateway sender instances together with a very small difference in time among them.

Reply:
I think that's the test case Gester mentioned in the comment as well.

Reply:
I understood he meant to stop the complete gateway sender, not just the primary.
It is not possible to solve all the problems if you do not stop in a short amount of time all instances. Either queues will not be drained if the primary is stopped but the secondary keeps receiving events after 15 seconds (with this solution) or the memory of the primary will grow indefinitely storing dropped events while it is not restarted (the latter would happen if we do not apply the fix proposed).